chroot_setup

```diff
@@ -0,0 +1,440 @@
+
+  if not TEMP_DIR.startswith('/QOpenSys/'):
+    print("\nERROR: [CHROOT DIRECTORY] Must begin with /QOpenSys/...\n")
+    parser.print_help
+    sys.exit(-2)
+
+  CHROOT_DIR = TEMP_DIR
+
+  if not os.path.isdir(CHROOT_DIR):
+    print(CHROOT_DIR + " does not exist.")
+    logging.debug(f"Creating directory at {CHROOT_DIR}")
+    os.mkdir(CHROOT_DIR)
+  else:
+    if not auto_yes:
+      print(CHROOT_DIR + " directory already exists")
+      val = input("Would you like to continue chroot setup? [y/N]")
+      if not val == "y":
+        print("Bye")
+        sys.exit(-4)
+
+
+  # Validate CHROOT TYPE Argument(s)
+  if not args.chroot_type:
+    # Only chroot_directory was provided to the script:
+    # Ask if minimal with includes chroot is desired
+    if not auto_yes:
+      val = input("Would you like to create a minimal chroot w/ includes @ " + CHROOT_DIR + "? [y/N]:")
+      if not val == "y":
+        print("Specify your desired [CHROOT TYPE]")
+        parser.print_help
+        sys.exit(-5)
+
+    mylist = []
+    mylist.append(SCRIPT_DIR + "/chroot_includes.lst")
+    mylist.append(SCRIPT_DIR + "/chroot_minimal.lst")
+
+  else:
+    # If chroot types were specified, then use those
+    for arg in args.chroot_type:
+      print("Chroot type is: " + arg)
+      
+      CHROOT_LIST = arg
+
+      # add chroot prefix if it doesn't exist
+      if not CHROOT_LIST.startswith('chroot'):
+        CHROOT_LIST = 'chroot_' + CHROOT_LIST
+
+      # check that the .lst file name was given
+      if not CHROOT_LIST.endswith('.lst'):
+        CHROOT_LIST = CHROOT_LIST + '.lst'
+
+      # check that the .lst file exists
+      if not os.path.isfile(CHROOT_LIST):
+        print(f"{CHROOT_LIST} cannot be found in {os.cwd}")
+        sys.exit(-7)
+      else:
+        # append file name to end of list
+        mylist.append(os.readlink(CHROOT_LIST))
+
+  logging.debug(f"Number of elements in the list: {len(mylist)}")
+  logging.debug(f"List contents: {mylist}")
+
+  # convert any global variables passed from a list of string "KEY=VALUE" to a
+  # dictionary holding "KEY": "VALUE"
+  globals_dictionary = {}
+  if args.globals:
+      for global_var in args.globals:
+        global_tokens = global_var.split('=')
+        globals_dictionary[global_tokens[0]] = global_tokens[1]
+
+  for chroot_file in mylist:
+    print("=====================================")
+    print("setting up based on " + chroot_file)
+    print("=====================================")
+
+    for key in globals_dictionary:
+      os.environ[key] = globals_dictionary[key]
+
+    chroot_setup(chroot_file)
+
+    for key in globals_dictionary:
+      del os.environ[key]
+
+  os.makedirs(args.chroot_directory + '/QOpenSys/etc', exist_ok=True)
+  etc_profile = (args.chroot_directory + '/QOpenSys/etc/profile')
+  if not os.path.isfile(etc_profile):
+    print('Priming /QOpenSys/etc/profile')
+    etc_file = open(etc_profile, 'a')
+    etc_file.write('PATH=/QOpenSys/pkgs/bin:$PATH\n')
```

Comment:
FYI, you can use print for this to add the newline

```suggestion
    print('PATH=/QOpenSys/pkgs/bin:$PATH', file=etc_file)
```

Reply:
fixed up in f8c5350b141afd53c324da0d4ccf2a3303661f08